addons/mail/static/tests/chat_window/chat_window_tests.js

```diff
@@ -1244,3 +1245,41 @@ QUnit.test(
         assert.containsOnce($, ".o-mail-ChatWindow");
     }
 );
+
+QUnit.test("hiding/swapping hidden chat windows does not update server state", async (assert) => {
+    patchUiSize({ size: SIZES.MD }); // only 2 chat window can be opened at a time
+    const pyEnv = await startServer();
+    pyEnv["discuss.channel"].create([{ name: "General" }, { name: "Sales" }, { name: "D&D" }]);
+    await start({
+        mockRPC(route, args) {
+            if (args.method === "channel_fold") {
+                const [channel] = pyEnv["discuss.channel"].searchRead([
+                    ["id", "=", args.args[0][0]],
+                ]);
+                assert.step(`${channel.name} - ${args.kwargs.state}`);
+            }
+        },
+    });
+    await clickContains(".o_menu_systray i[aria-label='Messages']");
+    await clickContains(".o-mail-NotificationItem", { text: "General" });
+    await contains(".o-mail-ChatWindow", { text: "General" });
+    assert.verifySteps(["General - open"]);
+    await clickContains(".o_menu_systray i[aria-label='Messages']");
+    await clickContains(".o-mail-NotificationItem", { text: "Sales" });
+    await contains(".o-mail-ChatWindow", { text: "Sales" });
+    assert.verifySteps(["Sales - open"]);
+    // Sales chat window will be hidden since there is not enough space for the
+    // D&D one but Sales fold state should not be updated.
+    await clickContains(".o_menu_systray i[aria-label='Messages']");
+    await clickContains(".o-mail-NotificationItem", { text: "D&D" });
+    assert.verifySteps(["D&D - open"]);
+    await contains(".o-mail-ChatWindow", { text: "D&D" });
```

Comment:
The step should be asserted after the channel is shown otherwise the check might come too early.